internal/v1/api.yaml

```diff
@@ -1601,6 +1601,9 @@ components:
           items:
             $ref: '#/components/schemas/CustomRepository'
           description: List of custom repositories.
+        template:
+          type: string
+          description: ID of template.
```

Comment:
I'd actually make this part of ImageRequest, not of its customisations.

Also maybe say that you can't have both a snapshot date and a template specified? Maybe even oneOf? Not sure.

Reply:
> Also maybe say that you can't have both a snapshot date and a template specified? Maybe even oneOf? Not sure.

Only if it makes sense to build images (and register systems) to particular snapshot dates? With introduction of templates, I think it makes sense to only restrict this to templates although technically snapshots are also repositories.

Also not sure, just food for thoughts.